pkg/controller/direct/bigqueryconnection/connection_controller.go

```diff
@@ -166,7 +166,14 @@ func (a *Adapter) Create(ctx context.Context, createOp *directbase.CreateOperati
 	if mapCtx.Err() != nil {
 		return mapCtx.Err()
 	}
-	status.ExternalRef = &created.Name
+
+	tokens := strings.Split(created.Name, "/")
+	parent, err = a.id.Parent()
+	if err != nil {
+		return err
+	}
+	externalRef := parent + "/connections/" + tokens[5]
```

Comment:
Another idea for future: maybe we should parse `created.Name` into a BigQueryConnectionConnectionRef, and then use the canonical form from BigQueryConnectionConnectionRef (I assume that's what we're doing here)

Reply:
Yes, agree! That's also how I want to refactor this code.